### PR TITLE
Add `as_mut_ptr` and inline hints

### PR DESCRIPTION
**Pull request by rahulchaphalkar**
- Added `as_mut_ptr()` and `as_ptr()` where required to return
- Added `#[inline]` compiler hints

### CODE REVIEW

File: crates/openvino/src/node.rs [hunk 1]
```diff
@@ -13,6 +13,7 @@ pub struct Node {
 
 impl Node {
     /// Create a new [`Port`] from [`ov_output_const_port_t`].
+    #[inline]
     pub(crate) fn new(ptr: *mut ov_output_const_port_t) -> Self {
```
Comment (abrown) [inline crates/openvino/src/node.rs:17]:
```suggestion
    pub(crate) fn from_ptr(ptr: *mut ov_output_const_port_t) -> Self {
```

For convention sake?

File: crates/openvino/src/request.rs [hunk 2]
```diff
@@ -20,12 +20,13 @@ unsafe impl Sync for InferRequest {}
...
 impl InferRequest {
     /// Create a new [`InferRequest`] from [`ov_infer_request_t`].
+    #[inline]
     pub(crate) fn from_ptr(ptr: *mut ov_infer_request_t) -> Self {
         Self { ptr }
     }
 
     /// Assign a [`Tensor`] to the input on the model.
-    pub fn set_tensor(&mut self, name: &str, tensor: &Tensor) -> Result<()> {
+    pub fn set_tensor(&mut self, name: &str, tensor: Tensor) -> Result<()> {
```
Comment (abrown) [inline crates/openvino/src/request.rs:29]: Can't remember why we need to own this tensor here?

**Review by abrown — APPROVED** [review]
Makes sense to me!

File: crates/openvino/src/model.rs [hunk 3]
```diff
@@ -22,11 +22,13 @@ unsafe impl Sync for Model {}
...
 impl Model {
     /// Create a new [`Model`] from an internal pointer.
+    #[inline]
     pub(crate) fn from_ptr(ptr: *mut ov_model_t) -> Self {
         Self { ptr }
     }
 
     /// Get the pointer to the underlying [`ov_model_t`].
+    #[inline]
     pub(crate) fn as_ptr(&self) -> *mut ov_model_t {
```
Comment (abrown) [inline crates/openvino/src/model.rs:32]:
```suggestion
    pub(crate) fn as_mut_ptr(&self) -> *mut ov_model_t {
```

File: crates/openvino/src/request.rs (hunk 2 above)
Comment (rahulchaphalkar) [reply crates/openvino/src/request.rs:29]: Done

**Pull request merged**